src/query/sql/src/planner/plans/plan.rs

```diff
@@ -302,123 +304,121 @@ pub enum RewriteKind {
     Call,
 }
 
-impl Display for Plan {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+impl Plan {
+    pub fn kind(&self) -> QueryKind {
```

Comment:
Notice that QueryKind is not accurate.

Some statements have been rewritten to Query like `ShowTables`

Reply:
Why not match the AST statement ?

Reply:
we can cache the  AST statement  in context.


```
pub(in crate::sessions) running_query_stmt: Statment,
```

Reply:
currently only used for copy.

Reply:
nn

> we can cache the AST statement in context.
> 
> ```
> pub(in crate::sessions) running_query_stmt: Arc<RwLock<Option<Statement>>>,,
> ```

need to be serialized and sent to other query nodes.

Reply:
also comparation is faster

Reply:
> nn
> 
> > we can cache the AST statement in context.
> > ```
> > pub(in crate::sessions) running_query_stmt: Arc<RwLock<Option<Statement>>>,,
> > ```
> 
> need to be serialized and sent to other query nodes.

Statement not impl Serialize for now, even if it does, it it too big for current usage.

Reply:
Ok, let's respect `RewriteKind` in query plan.

Reply:
QueryKind is from Plan,  those stmt in RewriteKind are already rewritten to Query at this point.